xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/objects/classes/ImplicitlyAllowedValuesDBListQueryBuilder.java

```diff
@@ -156,8 +174,17 @@ private void addFieldToQuery(String fieldName, String fieldAlias, boolean hasCla
         } else if (!hasClassName) {
             selectClause.add(DOC_PREFIX + fieldName);
         } else {
+            PropertyClass pc = null;
+            try {
+                pc = (PropertyClass) contextProvider.get().getWiki()
+                    .getXClass(documentReferenceResolver.resolve((String) parameters.get(CLASS_NAME)),
```

Comment:
What is the "current" document in this case? The class reference is probably local, so the wiki part needs to be added. I see above:

    query.setWiki(spec.wiki);

and

    spec.wiki = dbListClass.getOwnerDocument().getDocumentReference().getWikiReference().getName();

So the query is not executed on the current wiki but on the wiki of the xclass that has the DBList property. Maybe you can replace the ``boolean hasClassName`` parameter with the full spec ``DBListQuerySpec`` in order to have quick access to ``hasClassName``, ``className`` and ``wiki``.